summarization/summarizer.py

```diff
@@ -356,3 +360,33 @@ def _get_stamp_descriptor_for_embedded_content(self, content):
         # type casting is required since the dict keys
         # are now of type str
         return self.embedded_descriptors_dict[str(content.content_type)]
+
+    def _collect_embedded_and_quoted_indices(self):
+        self.embedded_indices = set([
+            content.content_index for content in self.contents.embedded_content
+        ])
+
+        self.quoted_indices = set([
+            content.content_index for content in self.contents.quoted_content
+        ])
+
+    def _get_stamp_type(self, stamp_page):
+        if stamp_page.media_index in self.embedded_indices:
+            return StampPageType.EMBEDDED
+
+        if stamp_page.media_index in self.quoted_indices:
+            return StampPageType.QUOTED
+
+        if stamp_page.media_index != -1:
+            if stamp_page.overlay_title and stamp_page.overlay_text:
+                return StampPageType.MEDIA_WITH_TEXT_AND_TITLE
+            elif stamp_page.overlay_title or stamp_page.overlay_text:
+                return StampPageType.MEDIA_WITH_TEXT
+            else:
+                return StampPageType.MEDIA_ONLY
+        else:
+            return StampPageType.TEXT_ONLY
+
+    def _set_stamp_page_types(self):
+        for stamp_page in self.stamp_pages_list:
+            stamp_page.stamp_type = self._get_stamp_type(stamp_page)
```

Comment:
Both of these methods should be in StampPage, and not here.

Reply:
It is possible to some degree but for determining whether it's an embedded content or quoted content would require the knowledge of the indices of the corresponding content. This is out of the StampPage object and so it's done here. 
Is there a better way to go about this?

Reply:
`stamp_page._get_stamp_type(self, embedded_indices, quoted_indices)`